include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -133,6 +133,41 @@ namespace concurrency {
 class ThreadPoolParallelSection;
 class ThreadPoolLoop;
 
+/* Usage:
+1. In executor, call Start() before profiling and Stop() to get profiled numbers;
+2. Inside thread pool, call LogStart() before interested section and LogEnd... after to log elapsed time;
+3. To extend, just add more events in enum Event before "All", and update GetEventName(...) accordingly.
+4. Note LogStart must pair with either LogEnd or LogEndAndStart, otherwise ORT_ENFORCE will fail.
+*/
+class ThreadPoolProfiler {
+ public:
+  enum ThreadPoolEvent {
+    DISTRIBUTION = 0,
+    DISTRIBUTION_ENQUEUE,
+    RUN,
+    WAIT,
+    WAIT_REVOKE,
+    MAX_EVENT
+  };
+  ThreadPoolProfiler();
+  ~ThreadPoolProfiler() = default;
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(ThreadPoolProfiler);
+  using Clock = std::chrono::high_resolution_clock;
+  inline bool Enabled() const;
+  void Start();                          //start profiling
+  std::string Stop();                    //stop profiling and return collected numbers
+  void LogStart();                       //record the starting time point
+  void LogEnd(ThreadPoolEvent);          //calculate and save the time elpased from last start point
+  void LogEndAndStart(ThreadPoolEvent);  //same as LogEnd but add a starting point before return
+
+ private:
+  inline const char* GetEventName(ThreadPoolEvent) const;
+  bool enabled_ = false;
+  std::thread::id main_thread_id_;
+  uint64_t events_[MAX_EVENT];
```

Comment:
>events_[MAX_EVENT] [](start = 11, length = 18)

consider `uint64_t events_[MAX_EVENT]{};`. then we don't need to define a constructor. alternatively, consider consistently initializing every member in the constructor member initializer list

Reply:
Let skip the constructor then.